### PR TITLE
[js-api] Tidy up rethrow identity test

### PR DESCRIPTION
**Pull request by aheejin**
This makes some cosmetic changes to the identity-preserving `rethrow` test added in #253. This is to prepare for adding the same set of tests for a Wasm-defined-and-exported tag, as requrested in https://github.com/WebAssembly/exception-handling/pull/253#issuecomment-1433703216.

### CODE REVIEW

**Comment by aheejin** [comment]
The tests using a Wasm-defined-and-exported tag will be added as a follow up after this lands. (Github doesn't support stacked PRs..)

File: test/js-api/exception/identity.tentative.any.js [hunk 1]
```diff
@@ -3,59 +3,68 @@
...
       kExprCatchAll,
         kExprRethrow, 0x00,
       kExprEnd
     ])
     .exportFunc();
 
   const buffer = builder.toBuffer();
+
+  // The exception object's identity should be preserved throw 'rethrow's in
+  // Wasm code. Do tests with a tag defined in JS.
```
Comment (Ms2ger) [inline test/js-api/exception/identity.tentative.any.js]: I'm finding it hard to follow the first sentence

File: test/js-api/exception/identity.tentative.any.js [hunk 2]
```diff
@@ -3,59 +3,68 @@
...
   WebAssembly.instantiate(buffer, imports).then(result => {
     try {
-      result.instance.exports.catch_rethrow();
+      result.instance.exports.catch_js_tag_rethrow();
     } catch (e) {
-      assert_equals(e, exn);
-      assert_not_equals(e, exn_same_payload);
-      assert_not_equals(e, exn_diff_payload);
+      assert_equals(e, jsTagExn);
+      assert_equals(e.getArg(jsTag, 0), jsTagExn.getArg(jsTag, 0));
```
Comment (Ms2ger) [inline test/js-api/exception/identity.tentative.any.js]: I don't see the value of this assertion; you already know they're the same object, so checking that the calling the same method on the same object returns the same result isn't very useful.

It might be interesting (if you hadn't planned that yet) to have a test where JS throws an exception (as you do here), and wasm catches it, but then returns the payload rather than using `rethrow`. Also one where wasm uses `throw` to create a new exception with the same payload.

**Review by Ms2ger — APPROVED** [review]
Thanks

File: test/js-api/exception/identity.tentative.any.js [hunk 3]
```diff
@@ -3,59 +3,68 @@
...
+        kExprCallFunction, throwJSTagExnIndex,
       kExprCatchAll,
         kExprRethrow, 0x00,
       kExprEnd
     ])
     .exportFunc();
 
   const buffer = builder.toBuffer();
+
+  // The exception object's identity should be preserved throw 'rethrow's in
```
Comment (dschuff) [inline test/js-api/exception/identity.tentative.any.js]:
```suggestion
  // The exception object's identity should be preserved across 'rethrow's in
```

File: test/js-api/exception/identity.tentative.any.js (hunk 2 above)
Comment (dschuff) [reply test/js-api/exception/identity.tentative.any.js]: Those seem like good ideas; the latter one could check both that the payload is the same and that the exception object is different.

**Review by dschuff — APPROVED** [review]

File: test/js-api/exception/identity.tentative.any.js (hunk 1 above)
Comment (aheejin) [reply test/js-api/exception/identity.tentative.any.js]: I committed @dschuff's suggestion: https://github.com/WebAssembly/exception-handling/pull/256#discussion_r1110065053 Is this clearer?

File: test/js-api/exception/identity.tentative.any.js (hunk 2 above)
Comment (aheejin) [reply test/js-api/exception/identity.tentative.any.js]: @Ms2ger 
> I don't see the value of this assertion; you already know they're the same object, so checking that the calling the same method on the same object returns the same result isn't very useful.

The checking of the payload was suggested in https://github.com/WebAssembly/exception-handling/pull/253#issuecomment-1433703216, but I agree it is not strictly necessary. I can delete it. @dschuff Are you OK with that?

> It might be interesting (if you hadn't planned that yet) to have a test where JS throws an exception (as you do here), and wasm catches it, but then returns the payload rather than using `rethrow`. Also one where wasm uses `throw` to create a new exception with the same payload.

Given that this PR intended only for cosmetic changes, do you mind if I do these in a follow-up?

Comment (dschuff) [reply test/js-api/exception/identity.tentative.any.js]: Yeah if we're going to have the object equality check, I agree we don't need the payload check. Followup change for more tests is fine with me.
